### PR TITLE
Add security headers and CSP report collection

### PR DESCRIPTION
**Pull request by gtanczyk**
## Summary
Implements comprehensive browser hardening headers for the API service, including Content-Security-Policy (CSP), framing protections, and a CSP violation reporting endpoint. This pins the security model documented in `docs/security-model.md`.

## Key Changes

- **New security headers module** (`security-headers.ts`):
  - Sets `X-Content-Type-Options: nosniff` and `Referrer-Policy: strict-origin-when-cross-origin` on all responses
  - Applies `Content-Security-Policy: frame-ancestors 'none'` and `X-Frame-Options: DENY` to HTML documents to prevent clickjacking
  - Adds `Permissions-Policy` restricting geolocation, payment, USB, and display capture
  - Implements report-only CSP for monitoring violations without enforcement
  - Respects routes that write their own CSP (e.g., sandboxed preview)

- **CSP violation reporting** (`/api/csp-report` endpoint):
  - Accepts both legacy `application/csp-report` and modern `application/reports+json` formats
  - Parses and summarizes reports, logging them at warn level
  - Rate-limited to 30 requests per minute to prevent abuse
  - Returns 204 No Content on success

- **Configuration**:
  - `APP_CSP_REPORT_ONLY` environment variable controls the report-only policy:
    - Unset or `"true"`: uses built-in policy
    - `"false"` or `"off"`: disables report-only header
    - Any other value: used verbatim as a draft policy for testing
  - Configurable via deployment variables and `.env.example`

- **Refactoring**:
  - Extracted canonical host redirect logic into `canonical-host.ts` for better organization
  - Registered security headers in `app.ts` after rate limiting (so CSP report endpoint is protected)

- **Comprehensive test coverage** (`security-headers.test.ts`):
  - Verifies headers on shell routes, 404 pages, and API endpoints
  - Tests OAuth consent screen protection
  - Validates CSP report acceptance and parsing
  - Tests configuration options and edge cases

## Notable Implementation Details

- Game frames inherit the report-only CSP but render from `blob:`/`srcdoc`, so inline script/style and `data:`/`blob:` media are allowed without triggering violations
- The hook respects routes that have already written a CSP header, allowing them to define their own embedding policy
- Report summarization handles both legacy and modern browser report formats, truncating large fields and gracefully handling malformed input
- The report-only policy is never enforced by this change; enforcement is a separate decision based on collected reports

https://claude.ai/code/session_01M1eLMLkVYbtZvEfikaeWN5

### CODE REVIEW

**Comment by gtanczyk** [comment]
@codex

File: docs/security-model.md [hunk 1]
```diff
@@ -96,6 +96,37 @@ Before publication, each game must pass the checks in
 consistency, size limit, credential scan, no remote dependencies, no frame-escape attempts, and
 a headless-browser smoke test. The gate complements review; it does not replace moderation.
 
+## Browser hardening headers
+
+One Cloud Run service serves the API and the web app, so response headers are set in one
+place: `apps/api/src/platform/security-headers.ts`, registered right after the cache policy in
+`app.ts`. Every response carries `X-Content-Type-Options: nosniff` and
```
[the remark this answers is not in the document]
Comment (gtanczyk) [reply docs/security-model.md]: Fixed in a4c0b9a: the doc now says the hook is registered right after the rate limiter (its report sink carries a rate-limit annotation).

---
_Generated by [Claude Code](https://claude.ai/code)_
